granulate_utils/metadata/databricks_client.py

```diff
@@ -44,45 +50,81 @@ def get_webui_address() -> Optional[str]:
     def get_job_name(self) -> Optional[str]:
         # Retry in case of a connection error, as the metrics server might not be up yet.
         for i in range(MAX_RETRIES):
-            time.sleep(10)
             try:
-                return self._get_job_name_impl()
-            except Exception:
-                self.logger.exception("Got Exception while collecting Databricks job name.")
+                if cluster_metadata := self._cluster_all_tags_metadata():
+                    # Got the job name, no need to retry.
+                    name = self._get_name_from_metadata(cluster_metadata)
+                    if name:
+                        self.logger.debug("Found name in metadata.", job_name=name, cluster_metadata=cluster_metadata)
+                        return name
+                    else:
+                        self.logger.debug("Failed to extract name from metadata.", cluster_metadata=cluster_metadata)
+                        return None
+                else:
+                    # No job name yet, retry.
+                    time.sleep(15)
+            except DatabricksJobNameDiscoverException as e:
+                self.logger.exception("Failed to get Databricks job name.", exception=e)
+                return None
+            except Exception as e:
+                self.logger.exception("Generic exception was raise during spark job name discovery.", exception=e)
+                return None
+        self.logger.info("Databricks get job name timeout, continuing...")
         return None
 
-    def _get_job_name_impl(self) -> Optional[str]:
-        # Make sure we're running on a databricks machine
-        if not os.path.isfile(DATABRICKS_DEPLOY_CONF_PATH):
-            return None
+    @staticmethod
+    def _get_name_from_metadata(metadata: Metadata) -> Optional[str]:
+        if "RunName" in metadata:
+            return str(metadata["RunName"]).replace(" ", "-").lower()
+        elif "Name" in metadata:
+            return str(metadata["Name"]).replace(" ", "-").lower()
+        elif "ClusterName" in metadata:
+            return str(metadata["ClusterName"]).replace(" ", "-").lower()
```

Comment:
Replace with a loop that tries the 3 potential values in this order, no need to duplicate the dict check / transformation etc...